lib/src/main/kotlin/net/dean/jraw/references/SubmissionReference.kt

```diff
@@ -82,6 +82,50 @@ class SubmissionReference internal constructor(reddit: RedditClient, id: String)
         }
     }
 
+    // TODO - Requires mod privileges for the subreddit that contains the post.
+    /**
+     * Sets the [spoilerStatus] of a post using the official Reddit spoiler flair.
+     */
+    @EndpointImplementation(Endpoint.POST_SPOILER, Endpoint.POST_UNSPOILER)
+    fun flagAsSpoiiler(spoilerStatus: Boolean) {
+        val endpoint = if (spoilerStatus) Endpoint.POST_SPOILER else Endpoint.POST_UNSPOILER
+        reddit.request {
+            it.endpoint(endpoint)
+                .post(mapOf(
+                    "id" to fullName
+                ))
+        }
+    }
+
+    // TODO - Requires mod privileges for the subreddit that contains the post.
+    /**
+     * Set or unset the announcement [state] of a post in its subreddit.
+     * Reddit provides 2 allowable announcement "slots" for a subreddit,
+     * often referred to as sticky posts.
+     *
+     * Including [num] will specify which sticky slot (of 2) to place the post into,
+     * if there is already an existing post in that slot, the new one will replace it.
```

Comment:
Currently Java users have to specify all three parameters, so excluding `num` isn't an option for them. Let's specify when `num` will be ignored.